<!DOCTYPE HTML>
<html>
<head>
  <meta charset="utf-8">
  
  <title>Hexo</title>

  
  <meta name="author" content="John Doe">
  

  

  

  <meta id="viewport" name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no, minimal-ui">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">

  

  <meta property="og:site_name" content="Hexo"/>

  
  <meta property="og:image" content="/favicon.ico"/>
  

  <link href="/favicon.ico" rel="icon">
  <link rel="alternate" href="/atom.xml" title="Hexo" type="application/atom+xml">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
<meta name="generator" content="Hexo 5.1.1"></head>


<body>
<div class="blog">
  <div class="content">

    <header>
  <div class="site-branding">
    <h1 class="site-title">
      <a href="/">Hexo</a>
    </h1>
    <p class="site-description"></p>
  </div>
  <nav class="site-navigation">
    <ul>
      
    </ul>
  </nav>
</header>

    <main class="site-main posts-loop">
    
  <article>

  
    
    <h3 class="article-title"><a href="/2021/03/07/硕士论文书写心得/"><span>硕士论文书写心得</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/2021/03/07/硕士论文书写心得/" rel="bookmark">
        <time class="entry-date published" datetime="2021-03-07T13:21:14.000Z">
          2021-03-07
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <p>  最近一直在忙着写硕士论文，刚刚写的时候特别顺手，总算有了一丝丝满意的感觉了。<br>我自己是十分珍惜书写硕士机会的。因为书写大论文的机会是不多的，如果将来不出书的情况下，即使读到博士，也就三次大论文机会。本科论文、硕士论文和博士论文,总共也就三次机会。所以即使有很多写小论文的机会，但是大论文的机会确实不多。并且在书写大论文的过程，对整个研究领域是有一个比较宏观和微观的梳理，最后心里会出现一条研究领域脉络的。</p>
<p>我在书写硕士论文背景的时候，我自己又重新梳理一遍转录组的文库制备和分析流程。对之前很多模模糊糊的概念重新有了较为清晰的认识。比如我一直只明白reads计数表面的意思，我在写的时候，就想在进行reads计数的时候，难道不是应该把reads拼接为完整的转录本，再进行计数吗？结果发现真不是，是直接统计数量的，那这样就要考虑测序深度和外显子长度了。所以归一化的时候，就要充分考虑这两方面的影响。</p>
<p>最近需要帮老师约转录组测序，还是很期待，比如茶树的测序深度多少呢？各家说的UMI绝对定量技术，真有那么神奇吗？</p>

      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    
    

    <span class="post-tags">
      <i class="icon-tags"></i>
        <a href="/tags/转录组/">转录组</a>
    </span>
    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/2021/01/31/NNL/"><span>NNL</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/2021/01/31/NNL/" rel="bookmark">
        <time class="entry-date published" datetime="2021-01-31T11:30:08.000Z">
          2021-01-31
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <p> 最近河南大学的王学路教授团队 (<a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41477-020-00832-7)%E5%9C%A8%E3%80%8ANature">https://www.nature.com/articles/s41477-020-00832-7)在《Nature</a> Plants》上报道他们新克隆一个新基因NNL1，转座子插入导致NNL1失活，失活的NNL1导致根瘤菌（rhizobia）侵染大豆根器官方式从裂缝侵染（crack infection）向根毛侵染（root hair infection）过渡。这说明NNL1是决定根瘤菌侵染方式的开关。</p>
<p>还有2020年最新发表的文章（<a target="_blank" rel="noopener" href="http://www.plantcell.org/content/32/6/1868%EF%BC%89%E3%80%82%E4%BD%9C%E8%80%85%E9%80%9A%E8%BF%87%E7%AA%81%E5%8F%98*Medicago">http://www.plantcell.org/content/32/6/1868）。作者通过突变*Medicago</a> truncatula* NODULE ROOT1 转换 legume-type nodules 为 actinorhizal-type nodules.</p>
<p>这不禁让我联想到《Actinorhizal root nodule symbioses: what is signalling telling on the origins of nodulation?》这篇文章。</p>
<p>到底信号通路告诉我们些什么呢？</p>

      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    
    

    <span class="post-tags">
      <i class="icon-tags"></i>
        <a href="/tags/根瘤进化-NNL-评论/">根瘤进化 NNL 评论</a>
    </span>
    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/2021/01/22/Stardock-Fence-报错/"><span>Stardock Fence 报错</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/2021/01/22/Stardock-Fence-报错/" rel="bookmark">
        <time class="entry-date published" datetime="2021-01-22T05:48:58.000Z">
          2021-01-22
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <h4 id="StarDock-Fence-报错"><a href="#StarDock-Fence-报错" class="headerlink" title="StarDock Fence 报错"></a>StarDock Fence 报错</h4><p>首先，因为课题和家里的一些自己的原因，我停更了好长时间，终于我终于又开始更新自己的博客了，感觉像回到家一样。</p>
<p>今天首先是解决Fence 软件的报错，我重装之后发现报错仍旧一样，我估计之前卸载不够干净，尤其是注册表。</p>
<p>之后使用注册表清理工具清理下无用的注册表之后，果然工作良好。</p>
<p>我好像很享受这种猜测的感觉？</p>

      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    
    

    <span class="post-tags">
      <i class="icon-tags"></i>
        <a href="/tags/Fence/">Fence</a>
    </span>
    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/2020/11/23/Kaks计算/"><span>Kaks计算</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/2020/11/23/Kaks计算/" rel="bookmark">
        <time class="entry-date published" datetime="2020-11-23T06:47:46.000Z">
          2020-11-23
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <p>  对于一个没有受到自然选择压力的基因来说，Ka/Ks 一般会小于等于1。但实际情况下，这个比值都是远小于1的，因为从蛋白质进化的角度而言，为了维持蛋白质功能的稳定性，其非同义替换的数量往往较少，因此非同义突变Ka要远远小于Ks。因此我们根据Ka/Ks大小，将基因划分为三类：</p>
<p>  Ka&gt;&gt;&gt;Ks -&gt; Ka/Ks &gt;&gt; 1, 基因受正选择 （positive selection）</p>
<p>  Ka = Ks -&gt; Ka/Ks = 1, 基因中性进化 （neutural selection）</p>
<p>  Ka&lt;&lt;&lt;Ks -&gt; Ka/Ks &lt;&lt; 1, 基因受纯化选择 （negative selection）</p>
<p>  对于一个基因是否受到正选择，我们不能简单的数一下两条序列之间发生（非）同义变换的SNP位点数，再除以（非）同义替换位点数，<strong>就得到Ka/Ks值！</strong>因为不同碱基之间发生替换的速率不同，比如转换（transition, 嘌呤&lt;=&gt;嘌呤,嘧啶&lt;=&gt;嘧啶）发生的概率要高于颠换（transversion, 嘌呤&lt;=&gt;嘧啶,嘧啶&lt;=&gt;嘌呤）；物种在不同进化支系进化速率也存在差异；这些都决定选择压力不仅要考虑基因序列差异，同时也要考虑基因的不同位点以及改基因所属的物种在进化中的进化速率来综合决定该基因是受何种选择。</p>
<p>Null hypothesis 零假设：所有lineage 有一个共同的w</p>
<p>Alternative hypothesis 替代假设：一些lineage 有共同的 w </p>
<p>不同的物种可能经历了不同的选择过程，因此 w值不同</p>
<p>如果替代假设正确，就可以从中推断进化的情况。</p>

      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    
    

    <span class="post-tags">
      <i class="icon-tags"></i>
        <a href="/tags/Ka-Ks计算/">Ka Ks计算</a>
    </span>
    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/2020/11/21/关于基因水平转移的思考/"><span>关于基因水平转移的思考</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/2020/11/21/关于基因水平转移的思考/" rel="bookmark">
        <time class="entry-date published" datetime="2020-11-21T13:15:38.000Z">
          2020-11-21
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        Here's something encrypted, password is required to continue reading.
        
      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    
    

    <span class="post-tags">
      <i class="icon-tags"></i>
        <a href="/tags/基因-水平转移/">基因 水平转移</a>
    </span>
    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/2020/11/11/Genome Aseembly/"><span></span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/2020/11/11/Genome Aseembly/" rel="bookmark">
        <time class="entry-date published" datetime="2020-11-10T19:23:43.325Z">
          2020-11-11
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <h4 id="Genome-Aseembly"><a href="#Genome-Aseembly" class="headerlink" title="Genome Aseembly"></a>Genome Aseembly</h4><h5 id="1-Download-data"><a href="#1-Download-data" class="headerlink" title="1.Download_data"></a>1.Download_data</h5><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR209/001/SRR2093871/SRR2093871_1.fastq.gz</span><br><span class="line">wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR209/001/SRR2093871/SRR2093871_2.fastq.gz</span><br><span class="line">wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR209/002/SRR2093872/SRR2093872_1.fastq.gz</span><br><span class="line">wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR209/002/SRR2093872/SRR2093872_2.fastq.gz</span><br><span class="line">wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR209/006/SRR2093876/SRR2093876_subreads.fastq.gz</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<h5 id="2-Assembly"><a href="#2-Assembly" class="headerlink" title="2.Assembly"></a>2.Assembly</h5><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">canu -p Bt2 -d Bt2_assembly genomeSize=6.2m  -pacbio-raw SRR2093876_subreads.fastq.gz </span><br></pre></td></tr></table></figure>

<h5 id="3-GenomeScope"><a href="#3-GenomeScope" class="headerlink" title="3. GenomeScope"></a>3. GenomeScope</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">jellyfish count -C -m 21 -s 1000000000 -t 10 *.fastq -o reads.jf</span><br><span class="line"></span><br><span class="line">jellyfish histo -t 10 reads.jf &gt; reads.histo</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<h5 id="4-Checking-your-Assembly-for-contaminants-using-VecScreen"><a href="#4-Checking-your-Assembly-for-contaminants-using-VecScreen" class="headerlink" title="4. Checking your Assembly for contaminants using VecScreen"></a>4. Checking your Assembly for contaminants using VecScreen</h5><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002/814/215/GCF_002814215.1_Sedor1/GCF_002814215.1_Sedor1_genomic.fna.gz</span><br></pre></td></tr></table></figure>

<h5 id="5-Genome-Scaffolding-with-pyScaf"><a href="#5-Genome-Scaffolding-with-pyScaf" class="headerlink" title="5. Genome Scaffolding with pyScaf"></a>5. Genome Scaffolding with pyScaf</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">git clone https:&#x2F;&#x2F;github.com&#x2F;lpryszcz&#x2F;pyScaf.git</span><br></pre></td></tr></table></figure>
      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/2020/11/06/Genome-Scope-study/"><span>Genome Scope study</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/2020/11/06/Genome-Scope-study/" rel="bookmark">
        <time class="entry-date published" datetime="2020-11-06T06:50:17.000Z">
          2020-11-06
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <p><img src="https://isugenomics.github.io/bioinformatics-workbook/assets/images/genomescope/screen_shot_2017-02-16_at_7.31.20_am.png" alt="GenomeScope"></p>
<p><img src="/2020/11/06/Genome-Scope-study/Users\Jinke\AppData\Roaming\Typora\typora-user-images\image-20201122214923389.png" alt="image-20201122214923389"></p>

      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/2020/11/03/重测序学习/重测序学习/"><span></span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/2020/11/03/重测序学习/重测序学习/" rel="bookmark">
        <time class="entry-date published" datetime="2020-11-03T10:48:20.724Z">
          2020-11-03
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <ol>
<li>SNP calling</li>
</ol>
<p><img src="/2020/11/03/%E9%87%8D%E6%B5%8B%E5%BA%8F%E5%AD%A6%E4%B9%A0/%E9%87%8D%E6%B5%8B%E5%BA%8F%E5%AD%A6%E4%B9%A0/Users\Jinke\AppData\Roaming\Typora\typora-user-images\image-20201103184910938.png" alt="image-20201103184910938"></p>

      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/2020/10/21/h5文件简介/"><span>h5文件简介</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/2020/10/21/h5文件简介/" rel="bookmark">
        <time class="entry-date published" datetime="2020-10-21T08:08:35.000Z">
          2020-10-21
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <p> h5文件是层次格式的</p>

      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    
    

    <span class="post-tags">
      <i class="icon-tags"></i>
        <a href="/tags/h5文件/">h5文件</a>
    </span>
    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/2020/10/19/matplotlib-学习/"><span>matplotlib 学习</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/2020/10/19/matplotlib-学习/" rel="bookmark">
        <time class="entry-date published" datetime="2020-10-19T13:16:06.000Z">
          2020-10-19
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <p> 使用 Matplotlib 提供的面向对象 API，需要导入 <code>pyplot</code> 模块，并约定简称为 <code>plt</code>。</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">from</span> matplotlib <span class="keyword">import</span> pyplot <span class="keyword">as</span> plt</span><br></pre></td></tr></table></figure>

<p><code>pyplot</code> 模块中 <code>pyplot.plot</code> 方法是用来绘制折线图的。你应该会很容易联想到，更改后面的方法类名就可以更改图形的样式。的确，在 Matplotlib 中，大部分图形样式的绘制方法都存在于 <code>pyplot</code> 模块中。例如：</p>
<table>
<thead>
<tr>
<th>方法</th>
<th>含义</th>
</tr>
</thead>
<tbody><tr>
<td><code>matplotlib.pyplot.angle_spectrum</code></td>
<td>绘制电子波谱图</td>
</tr>
<tr>
<td><code>matplotlib.pyplot.bar</code></td>
<td>绘制柱状图</td>
</tr>
<tr>
<td><code>matplotlib.pyplot.barh</code></td>
<td>绘制直方图</td>
</tr>
<tr>
<td><code>matplotlib.pyplot.broken_barh</code></td>
<td>绘制水平直方图</td>
</tr>
<tr>
<td><code>matplotlib.pyplot.contour</code></td>
<td>绘制等高线图</td>
</tr>
<tr>
<td><code>matplotlib.pyplot.errorbar</code></td>
<td>绘制误差线</td>
</tr>
<tr>
<td><code>matplotlib.pyplot.hexbin</code></td>
<td>绘制六边形图案</td>
</tr>
<tr>
<td><code>matplotlib.pyplot.hist</code></td>
<td>绘制柱形图</td>
</tr>
<tr>
<td><code>matplotlib.pyplot.hist2d</code></td>
<td>绘制水平柱状图</td>
</tr>
<tr>
<td><code>matplotlib.pyplot.pie</code></td>
<td>绘制饼状图</td>
</tr>
<tr>
<td><code>matplotlib.pyplot.quiver</code></td>
<td>绘制量场图</td>
</tr>
<tr>
<td><code>matplotlib.pyplot.scatter</code></td>
<td>散点图</td>
</tr>
<tr>
<td><code>matplotlib.pyplot.specgram</code></td>
<td>绘制光谱图</td>
</tr>
</tbody></table>
<p><code>matplotlib.pyplot.plot(*args, **kwargs)</code> 方法严格来讲可以绘制线形图或者样本标记。其中，<code>*args</code> 允许输入单个 <em>𝑦</em> 值或 <em>𝑥</em>,<em>𝑦</em> 值</p>

      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    
    

    <span class="post-tags">
      <i class="icon-tags"></i>
        <a href="/tags/matplotlib-指引/">matplotlib 指引</a>
    </span>
    

    </div>

    
  </div>
</article>




<nav class="pagination">
  
  
  <a href="/page/2/" class="pagination-next">Next</a>
  
</nav>
    </main>

    <footer class="site-footer">
  <p class="site-info">
    Proudly powered by <a href="https://hexo.io/" target="_blank">Hexo</a> and
    Theme by <a href="https://github.com/CodeDaraW/Hacker" target="_blank">Hacker</a>
    </br>
    
    &copy; 2021 John Doe
    
  </p>
</footer>
    
  </div>
</div>
</body>
</html>